src/Network/Ethereum/Web3/TH.hs

```diff
@@ -265,7 +265,7 @@ mkFun fun@(DFunction name constant inputs outputs) = (++)
         ]
   where mIdent    = T.unpack (methodId $ fun{funName = T.replace "'" "" name})
         dataName  = mkName (toUpperFirst (T.unpack $ name <> "Data"))
-        funName   = mkName (toLowerFirst (T.unpack name))
+        funName   = mkName (toLowerFirst (T.unpack name) ++ "F")
```

Comment:
Please explain this addition.

Reply:
Unfortunately it was to avoid a collision in even a simple ERC20 contract. `transferTo` is a function, and `transferTo` is also a record accessor for the `Transfer` event type. 

I don't know if there is a particularly good way of dealing with this problem, but this works and it's at least uniform.

Reply:
It seems like function in contract with name `foo(bool y) constant returns (bool)` will be available as `fooF :: Bool -> Web3 p Bool`, right? In my opinion it's not good practice, may be we can keep selectors as is, but for collision escaping prefix/postfix event records by some character like http://solidity.readthedocs.io/en/develop/style-guide.html#avoiding-naming-collisions for event arguments.